### PR TITLE
Slice buffer in hash_func macro, instead of in all callers

### PR DESCRIPTION
**Pull request by Dr-Emann**

### CODE REVIEW

File: src/hashing/crc8.rs [hunk 1]
```diff
@@ -3,5 +3,5 @@ use crc8::Crc8;
 
 
 hash_func!((Crc8::create_lsb(0x9b), 0u8),
-           |state_crc: &mut (Crc8, u8), buffer: &[u8], read: usize| state_crc.1 = state_crc.0.calc(buffer, read as i32, state_crc.1),
+           |state_crc: &mut (Crc8, u8), buffer: &[u8]| state_crc.1 = state_crc.0.calc(buffer, buffer.len() as i32, state_crc.1),
```
Comment (Dr-Emann) [inline src/hashing/crc8.rs:6]: This should be equivalent, since buffer is now sliced to the correct length.

**Comment by nabijaczleweli** [comment]
:+1: :shipit: What's with the mode changes, though `(100644 → 100755)`?

**Comment by Dr-Emann** [comment]
Hmm. Probably what I get for opening editing in windows from a linux share.

**Comment by nabijaczleweli** [comment]
What's `x` on normal text files useful for anyway?

**Comment by nabijaczleweli** [comment]
Reverted the mode changes

**Pull request merged**